### PR TITLE
Focus and student focus models

### PR DESCRIPTION
**Pull request by KNAthari**
### Changes
Give a short summary of the changes made in this PR. 
Generated models and wrote test for Focus and Studentfocus models. Revisited code and implemented feedback to code.

### Link to Issue 
(https://app.asana.com/0/1206510424537042/1206833138919401/f)

### Comments, Notes, or  Questions
This section is if you want to share something related to the PR with the reviewer. 

### Related Screenshots 
For FE tickets specifcially, please include a screenshot of your work!

### Before you submit for review, Did you...

- [x] Added relevant tests (if applicable)
- [x] Linked Asana ticket to PR
- [x] Added comments, notes, or questions for reviewwer
- [x] Added relevant screenshots
- [x] Ensured ticket title is clear and concise(i.e. 'Donation Form Feature' instead of 'sfg/chels/donation-form'

### CODE REVIEW

File: db/seeds.rb [hunk 1]
```diff
@@ -1,10 +1,6 @@
...
-KnowingCategory.find_or_create_by(name: 'Interest', descriptions: 'Interests/Passions', color: '#32b021', active: true)
-KnowingCategory.find_or_create_by(name: 'Talents', descriptions: 'Talents/Skills', color: '#0c87cf', active: true)
-KnowingCategory.find_or_create_by(name: 'Dislikes', descriptions: 'Dislikes/Challenges', color: 'ffba08', active: true)
-KnowingCategory.find_or_create_by(name: 'Family', descriptions: 'Family/Home', color: '#642ca9', active: true)
-KnowingCategory.find_or_create_by(name: 'Friends', descriptions: 'Friends/Peers', color: '#f75c03', active: true)
-KnowingCategory.find_or_create_by(name: 'Beliefs', descriptions: 'Beliefs/Thoughts', color: '#d00000', active: true)
+if Rails.env.development?
+  load(Rails.root.join('db/seeds/classroom_setup.rb'),
+       Rails.root.join('db/seeds/focus_setup.rb'))
+end
```
Comment (edithemmings) [inline db/seeds.rb:6]: Looks like you might have some conflicts in this file. Make sure you aren't deleting others' seeds!

**Review by edithemmings — APPROVED** [review]
Looks great! Just make sure you resolve any conflicts you have with the main branch. Awesome work!!

File: db/schema.rb [hunk 2]
```diff
@@ -103,8 +124,12 @@
 
   add_foreign_key "classrooms", "schools"
   add_foreign_key "educators", "schools"
-  add_foreign_key "educators", "users"
```
Comment (edithemmings) [inline db/schema.rb:106]: What happened here? I dont' think we want to remove this

File: db/schema.rb [hunk 3]
```diff
@@ -48,7 +49,7 @@
     t.index ["school_id"], name: "index_educators_on_school_id"
     t.index ["user_id"], name: "index_educators_on_user_id"
   end
-
+
```
Comment (edithemmings) [inline db/schema.rb:52]: Extra space

File: db/schema.rb (hunk 2 above)
Comment (KNAthari) [reply db/schema.rb:106]: I put it back

File: db/migrate/20240325173651_create_student_focuses.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class CreateStudentFocuses < ActiveRecord::Migration[7.1]
+  def change
+    create_table :student_focuses do |t|
+      t.references :focus, null: false, foreign_key: true
+      t.references :student, null: false, foreign_key: true
+      t.references :educator, foreign_key: true
```
Comment (edithemmings) [inline db/migrate/20240325173651_create_student_focuses.rb:8]: I think you actually _do_ want `null: false` here. A student_focus should always belong_to an educator

File: app/models/focus.rb [hunk 5]
```diff
@@ -0,0 +1,6 @@
+# frozen_string_literal: true
+
+class Focus < ApplicationRecord
+  validates :name, presence: true
```
Comment (edithemmings) [inline app/models/focus.rb:4]: This should probably be accompanied by a `null: false` on the database table column

**Comment by KNAthari** [comment]
> Looks like you need to rollback and rerun your migration since the null:false addition was not picked up in the schema

So, I ran the migrations and an issue appeared. One of the test is failing, here is the message: 'ActiveRecord::NotNullViolation: PG::NotNullViolation: ERROR:  null value in column "educator_id" of relation "student_focuses" violates not-null constraint,' I think this is in relation to the user model that was merged yesterday.

**Review by edithemmings — APPROVED** [review]
🎉 Great work!! Thanks for addressing all the comments

**Pull request merged**